src/adaptors/http2/http2_adaptor.c

```diff
@@ -2941,15 +2941,9 @@ void qd_http2_delete_listener(qd_dispatch_t *qd, qd_http_listener_t *li)
 /**
  * Create listener via Management request
  */
-qd_http_listener_t *qd_http2_configure_listener(qd_dispatch_t *qd, qd_http_adaptor_config_t *config, qd_entity_t *entity)
+qd_http_listener_t *qd_http2_configure_listener(qd_http_listener_t *li, qd_dispatch_t *qd, qd_entity_t *entity)
 {
-    qd_http_listener_t *li = qd_http_listener(qd->server, config);
-    if (!li) {
-        qd_log(http2_adaptor->log_source, QD_LOG_ERROR, "Unable to create http listener: no memory");
-        return 0;
-    }
-
-    li->adaptor_listener = qd_adaptor_listener(qd, config->adaptor_config, http2_adaptor->log_source);
+    li->adaptor_listener = qd_adaptor_listener(qd, li->config->adaptor_config, http2_adaptor->log_source);
```

Comment:
On an *unrelated* note, I was looking into why each adaptor has its own listener like say ```qd_tcp_listener``` and then they share a common listener ```qd_adaptor_listener_t```
All adaptors ultimately call ```qd_adaptor_listener_listen()```
Why can't all adaptors just use ```qd_adaptor_listener_t``` ?
Is that because of the ```protocolVersion``` field in the httpConnector/httpListener ? What if we replaced say httpListener with ```http1Listener``` and ```http2Listener``` and remove the ```protocolVersion``` field ? Is that going to be a problem when tomorrow if `http3` comes along ?

Reply:
I would keep them separate personally because they are concerned with separate issues.  The qd_adaptor_listener_t isn't a complete listener object - it only manages the lifecycle of the associated pn_listener_t socket.  A qd_(tcp|http)_listener_t has-a qd_adaptor_listener_t, but isn't derived from one.

Possibly the qd_adaptor_listener_t name is misleading since it isn't a representation of a tcp or http listener.  I'm lousy at names, but something involving "proton listener socket lifecycle manager" would be a more accurate description.  Or qd_listener_socket_manager_t maybe....